src/control/server/config.go

```diff
@@ -93,37 +93,35 @@ func (c *configuration) setPath(path string) error {
 
 // loadConfigOpts derives file location and parses configuration options
 // from both config file and commandline flags.
-func loadConfigOpts(cliOpts *cliOptions, host string) (
-	config configuration, err error) {
-
-	config = newConfiguration()
+func loadConfigOpts(cliOpts *cliOptions, host string) (*configuration, error) {
+	config := newConfiguration()
 
 	if err := config.setPath(cliOpts.ConfigPath); err != nil {
-		return config, errors.WithMessage(err, "set path")
+		return &config, errors.WithMessage(err, "set path")
 	}
 
 	if err := config.loadConfig(); err != nil {
-		return config, errors.Wrap(err, "read config file")
+		return &config, errors.Wrap(err, "read config file")
```

Comment:
Shouldn't these be returning nil, errors.Wrap(err, ...)? If failing to perform these steps are errors, then returning a config after these failures could result in undefined behavior at some point if the caller doesn't handle errors as you expect here.